test/conftest.py

```diff
@@ -110,3 +110,92 @@ def test_parquet_file(tmpdir, test_csv_file) -> str:
         pq.write_table(model_output_table, test_parquet_file)
 
     return str(test_file_path)
+
+
+#
+# schema-related fixtures that patch schema-related functions to override a pa.schema
+#
+
+@pytest.fixture()
+def schema_empty(mocker):
+    schema = pa.schema([])
+    mocker.patch("hubverse_transform.model_output.ModelOutputHandler._read_tasks", return_value={'a': 'dict'})
+    mocker.patch("hubverse_transform.model_output.ModelOutputHandler._get_schema", return_value=schema)
+
+
+@pytest.fixture()
+def schema_reference_date(mocker):
+    schema = pa.schema([('reference_date', pa.date32()),
+                        ('target', pa.string()),
+                        ('horizon', pa.int64()),
+                        ('target_end_date', pa.date32()),
+                        ('location', pa.string()),
+                        ('output_type', pa.string()),
+                        ('output_type_id', pa.string()),
+                        ('value', pa.float64()),
+                        ('round_id', pa.string()),
+                        ('model_id', pa.string())])
+    mocker.patch("hubverse_transform.model_output.ModelOutputHandler._read_tasks", return_value={'a': 'dict'})
+    mocker.patch("hubverse_transform.model_output.ModelOutputHandler._get_schema", return_value=schema)
+
+
+@pytest.fixture()
+def schema_origin_date(mocker):
+    schema = pa.schema([('origin_date', pa.date32()),
+                        ('target', pa.string()),
+                        ('horizon', pa.int64()),
+                        ('location', pa.string()),
+                        ('output_type', pa.string()),
+                        ('output_type_id', pa.string()),
+                        ('value', pa.float64()),
+                        ('round_id', pa.date32()),
+                        ('model_id', pa.string())])
+    mocker.patch("hubverse_transform.model_output.ModelOutputHandler._read_tasks", return_value={'a': 'dict'})
+    mocker.patch("hubverse_transform.model_output.ModelOutputHandler._get_schema", return_value=schema)
+
+
+@pytest.fixture()
+def schema_origin_date_str_val(mocker):
+    schema = pa.schema([('origin_date', pa.date32()),
+                        ('target', pa.string()),
+                        ('horizon', pa.int64()),
+                        ('location', pa.string()),
+                        ('output_type', pa.string()),
+                        ('output_type_id', pa.string()),
+                        ('value', pa.string()),  # only diff b/w this and schema_origin_date()
```

Comment:
Quick question. Why are we casting `value` as string when it can only ever be int or double?  Just to check that the casting works?

Reply:
That's required by the test `test_missing_model_output_id_mixture()`, which reads the file [2024-07-07-teamabc-output_type_ids_mixed.csv](https://github.com/hubverse-org/hubverse-transform/blob/main/test/integration/data/2024-07-07-teamabc-output_type_ids_mixed.csv) that contains non-numeric `value` rows:

> E   pyarrow.lib.ArrowInvalid: In CSV column #6: CSV conversion error to double: invalid value '0.1.654'